x-pack/plugins/security/public/management/api_keys/api_keys_api_client.ts

```diff
@@ -5,39 +5,60 @@
  */
 
 import { HttpStart } from 'src/core/public';
-import { ApiKey, ApiKeyToInvalidate } from '../../../common/model';
+import { ApiKey, ApiKeyToInvalidate, Role } from '../../../common/model';
 
-interface CheckPrivilegesResponse {
+export interface CheckPrivilegesResponse {
   areApiKeysEnabled: boolean;
   isAdmin: boolean;
   canManage: boolean;
 }
 
-interface InvalidateApiKeysResponse {
+export interface InvalidateApiKeysResponse {
   itemsInvalidated: ApiKeyToInvalidate[];
   errors: any[];
 }
 
-interface GetApiKeysResponse {
+export interface GetApiKeysResponse {
   apiKeys: ApiKey[];
 }
 
+export interface CreateApiKeyRequest {
+  name: string;
+  expiration?: string;
+  role_descriptors?: {
+    [key in string]: Role['elasticsearch'];
+  };
+}
+
+export interface CreateApiKeyResponse {
+  id: string;
+  name: string;
+  expiration: number;
+  api_key: string;
+}
+
 const apiKeysUrl = '/internal/security/api_key';
 
 export class APIKeysAPIClient {
   constructor(private readonly http: HttpStart) {}
 
-  public async checkPrivileges() {
+  public checkPrivileges = async () => {
```

Comment:
For my own education: was there a particular reason to change from `public async function() {}` to `public function = async () => {}`?

Do you just prefer the arrow functions, or did the implicit `this` binding benefit you in some way?